Xamarin.Forms.Platform.Android/AppCompat/MasterDetailContainer.cs

```diff
@@ -61,24 +60,10 @@ protected override void AddChildView(VisualElement childView)
 
 			Page page = childView as NavigationPage ?? (Page)(childView as TabbedPage);
 
+			// The thing we're adding is not a NavigationPage or TabbedPage, so we can just use the old AddChildView 
 			if (page == null)
 			{
-				// The thing we're adding is not a NavigationPage or TabbedPage, so we can just use the old AddChildView 
-
-				if (_currentFragment != null)
-				{
-					// But first, if the previous occupant of this container was a fragment, we need to remove it properly
-					FragmentTransaction transaction = FragmentManager.BeginTransaction();
-					transaction.Remove(_currentFragment);
-					transaction.SetTransition((int)FragmentTransit.None);
-
-					// This is a removal of a fragment that's not going on the back stack; there's no reason to care
-					// whether its state gets successfully saved, since we'll never restore it. Ergo, CommitAllowingStateLoss
-					transaction.CommitAllowingStateLoss();
-
-					_currentFragment = null;
-				}
```

Comment:
This whole block can be refactored into a method since similar logic is being used while disposing the container. Note that this block was NOT calling `ExecutePendingTransactions()` whereas the refactored method IS. I'm not sure if this can create any issues. @hartez removed forced execution of pending transactions in #8771b4fa for navigation pages. Let me know if this needs to change back to the original logic.